daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs

```diff
@@ -672,23 +680,57 @@ runStart sandboxPort (StartNavigator shouldStartNavigator) (OpenBrowser shouldOp
                   then withNavigator
                   else (\_ _ _ f -> f sandboxPh)
 
-runDeploy :: String -> SandboxPort -> IO ()
-runDeploy host sandboxPort = do
+data HostAndPortFlags = HostAndPortFlags { hostM :: Maybe String, portM :: Maybe Int }
+
+withHostAndPortDefaults :: HostAndPortFlags -> IO HostAndPort
+withHostAndPortDefaults HostAndPortFlags{hostM,portM} = do
+    defaultHost <- getProjectLedgerHost
+    defaultPort <- getProjectLedgerPort
+    let host = fromMaybe defaultHost hostM
+    let port = fromMaybe defaultPort portM
```

Comment:
In both of these cases, you only really want to get the default host and port if they weren't specified, right? I think `fromMaybeM getProjectLedgerHost hostM` and `fromMaybeM getProjectLedgerPort portM` do what we want here.

Reply:
ok!

Reply:
done